openmp/libomptarget/plugins-nextgen/common/src/PluginInterface.cpp

```diff
@@ -1492,7 +1492,7 @@ Error GenericPluginTy::init() {
   GlobalHandler = createGlobalHandler();
   assert(GlobalHandler && "Invalid global handler");
 
-  RPCServer = new RPCServerTy(NumDevices);
+  RPCServer = new RPCServerTy();
   assert(RPCServer && "Invalid RPC server");
```

Comment:
It's not safe to use `assert` for runtime check since in release mode this is simply gone. Potentially if the pointer is null, return failure.

Reply:
This is just a sanity check in-line with the ones above it. I'm not really too concerned with whether or not `new` runs out of memory in a full build so it's not worth checking since it's not like we could handle it.

Reply:
If we assume later that this pointer will not be null, then we really should take care of it here; otherwise we don't need the check here.

Reply:
There's a ridiculous amount of  `std::make_unique<T>()`  which we all just assume will not be null. I don't think our library is capable of handling the allocator failing as an expected failure mode. But maybe we could update this in the future. However I think it's out of scope for now.